### PR TITLE
Disable everything except restore / hard delete in trash

### PR DESCRIPTION
**Pull request by MrFlashAccount**
### Pull Request Description

Closes: enso-org/cloud-v2#1084

Tl;dr: This PR disable almost all actions in trash folder, except restore/hard delete.
Context:
Currently, we can do a lot of actions it trash folder: arrange files in folders, create new folders, edit files and so on. We want to disable most of these actions.

This Change:

- Disables d&d support within trash folder, but leaves the ability to drag thigs back to home folder.
- Disables most of the keyboard shortcuts(copy,paste,new...etc) except hard delete/restore
- Disables launching projects
- Disables rename
- Disables assigning labels
- Disabled editing the description & sharing

https://github.com/enso-org/enso/assets/61194245/35c6532d-719f-46b2-a2f7-b54b54856bbd

Test Plan:

This PR expects thorough testing to make sure we have disabled everything except hard delete/restore and the same way we didn't break the home folder :)

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [ ] The documentation has been updated, if necessary.
- [ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [ ] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [ ] Unit tests have been written where possible.
  - [ ] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

**Comment by PabloBuchu** [comment]
Very minor thing probably not related. The project icon in the trash is weird. It should be the same `play` button but not clickable

Screenshot 2024-04-03 at 14 09 47

**Comment by MrFlashAccount** [comment]
@PabloBuchu 
> Very minor thing probably not related. The project icon in the trash is weird. It should be the same play button but not clickable

I can't entirely agree, on the one hand, the `play` button is smth that is associated with a project, but on the other hand, we might confuse our users when they see the `play` button which does nothing. I think it is worth distinguishing between projects users can launch and those they can't launch.

Also, I agree that the icon isn't the best.

File: app/ide-desktop/lib/dashboard/src/layouts/AssetsTable.tsx [hunk 1]
```diff
@@ -1858,9 +1858,9 @@ export default function AssetsTable(props: AssetsTableProps) {
     }
   }
 
-  const state = React.useMemo(
+  const state = React.useMemo<AssetsTableState>(
     // The type MUST be here to trigger excess property errors at typecheck time.
```
Comment (somebody1234) [inline app/ide-desktop/lib/dashboard/src/layouts/AssetsTable.tsx:1882]: should probably move the comment

**Review by somebody1234 — APPROVED** [review]
approving code changes

**Review by PabloBuchu — APPROVED** [review]

**Comment by PabloBuchu** [comment]
Approved we can disagree on an icon somewhere elese
